src/Chapter3.hs

```diff
@@ -343,6 +344,31 @@ Define the Book product data type. You can take inspiration from our description
 of a book, but you are not limited only by the book properties we described.
 Create your own book type of your dreams!
 -}
+data Date = Date
+    { day   :: Int  -- is there a way to restrict acceptable values?
+    , month :: Int
+    , year  :: Int
+    }
+
+data Author = Author
+    { firstName  :: String
+    , lastName   :: String
+    , middleName :: String -- is there a way to make this optional?
```

Comment:
Yes, you can use the `Maybe` data type here. We speak a bit about that type later in the chapter, but it exactly means that it either exists or not 🙂 
```suggestion
    , middleName :: Maybe String
```